src/helpers/import-helper.js

```diff
@@ -0,0 +1,35 @@
+async function supportsDynamicImport() {
+  try {
+    // imports are cached.
+    // no need to worry about perf here.
+    // Don't remove .js: extension must be included for ESM imports!
+    await import('./dummy-file.js');
+    return true;
+  } catch (e) {
+    return false;
+  }
+}
+
+/**
+ * Imports a JSON, CommonJS or ESM module
+ * based on feature detection.
+ *
+ * @param modulePath path to the module to import
+ * @returns {Promise<unknown>} the imported module.
+ */
+async function importModule(modulePath) {
+  // JSON modules are still behind a flag. Fallback to require for now.
+  // https://nodejs.org/api/esm.html#json-modules
+  if (!modulePath.endsWith('.json') && (await supportsDynamicImport())) {
+    return import(modulePath);
+  }
+
+  // mimics what `import()` would return for
+  // cjs modules.
+  return { default: require(modulePath) };
```

Comment:
This assumes that the module only has a default export, and is not compatible with named exports.

For the purposes of the config that's fine, because the config is exported on the default export when is ESM compatible, but this is not true for all cases.

Which makes me feel like it would be better to invoke `supportsDynamicImport` in `config-helper.js` and either use the import helper, or straight up require.

Reply:
I'm not sure I understand what you mean.

If the user is using named exports then the platform is compatible with ESM and `import()` will be used, which supports named exports.

If they're using CJS, there are no named-exports. 

Native `import()` of a CJS file already returns `{ default: module.exports }`.

Reply:
I understand what you mean, I think I didn't fully understand what you were saying in #982 

But basically you wanted to write a node 10 compatible `importModule` function, while I was under the impression that you were going to drop support for node 10.

Reply:
We'll indeed drop Node 10 for v7, but I don't think we want to wait with fixing this until the core sequelize package is ready for that

Reply:
Yea, makes sense, I think we can resolve this then.

Reply:
Great, so that's an approval from you? Then I think we can merge and set up a new release

Reply:
Yes exactly :) The point was to replicate `import()` using `require`, the code that uses `import-helper` is then responsible for properly using the module

Reply:
> The point was to replicate import() using require

For sequelize v7, will this solution be dropped and using the native `import` be supported instead? Along with `.sequelizerc` no longer having `module.exports` ? Should I open an issue for this?

Reply:
Yes, the next major will drop support for node 10 so `import()` will always be used.

Supporting ESM in `.sequelizerc` could already be done in this version though. It should be a small change too.
Simply need to make this line use the new import helper https://github.com/sequelize/cli/blob/54ec29d65727873f88d0daa50e5e92ef7d00a4b8/src/core/yargs.js#L9 + support searching for `.sequelizerc` + `.sequelizerc.json` + `.sequelizerc.cjs` + `.sequelizerc.mjs` + `.sequelizerc.js`.

(feel free to open an issue)

Reply:
Thanks! I've openeded https://github.com/sequelize/cli/issues/989 , which is only for v7.
I think current version is fine as is, would prefer focusing on v7 since sequelize v7 is almost ready.